api/v1beta1/azurecluster_default.go

```diff
@@ -245,6 +246,29 @@ func (c *AzureCluster) setAPIServerLBDefaults() {
 				},
 			}
 		}
+		// if the API Server ILB feature is enabled, we should create a default internal LB IP
```

Comment:
These are not the changes we want.

We want the new feature flag to gate these code paths:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-b74c6acb4629abe58f8286da9939563cc4404e09fe9966eb61dbde9b87274189R268
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-b6931ac61f37a13f66fc7836e039bf714a0553d5db4f313cc8ef31ca5f5647b9R301
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-e05ece175e7c4d07287224549778fa3f2c09deed6e6b3f5bfa95fed5ab18cfbfR217
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-e05ece175e7c4d07287224549778fa3f2c09deed6e6b3f5bfa95fed5ab18cfbfR258
-

Reply:
Commit 90ba0f0deca685d0cc934c37b5510e363eb1798b adds feature flag for 
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-b74c6acb4629abe58f8286da9939563cc4404e09fe9966eb61dbde9b87274189R268
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-b6931ac61f37a13f66fc7836e039bf714a0553d5db4f313cc8ef31ca5f5647b9R301

Below commits do not need a feature flag tho.
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-e05ece175e7c4d07287224549778fa3f2c09deed6e6b3f5bfa95fed5ab18cfbfR217
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209/files#diff-e05ece175e7c4d07287224549778fa3f2c09deed6e6b3f5bfa95fed5ab18cfbfR258

Reply:
> The changes to enable configurable apiserver ILB can come later.
Ok. 

This PR is ready too. 
But I see the suggestion is just "gate" the code changes, so splitting this PR into two.